<?xml version="1.0" encoding="utf-8"?>
<local:TryCatchWorkflow mva:VisualBasic.Settings="Assembly references and imported namespaces serialized as XML namespaces" xmlns="http://schemas.microsoft.com/netfx/2009/xaml/activities" xmlns:e="clr-namespace:ExceptionWorkflow;assembly=ExceptionWorkflow, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null" xmlns:local="clr-namespace:ExceptionWorkflow;assembly=ExceptionWorkflow" xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006" xmlns:mv="clr-namespace:Microsoft.VisualBasic;assembly=System" xmlns:mva="clr-namespace:Microsoft.VisualBasic.Activities;assembly=System.Activities" xmlns:s="clr-namespace:System;assembly=mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089" xmlns:s1="clr-namespace:System;assembly=mscorlib" xmlns:s2="clr-namespace:System;assembly=System" xmlns:s3="clr-namespace:System;assembly=System.Xml" xmlns:s4="clr-namespace:System;assembly=System.Core" xmlns:sa="clr-namespace:System.Activities;assembly=System.Activities, Version=4.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35" xmlns:sad="clr-namespace:System.Activities.Debugger;assembly=System.Activities" xmlns:sap="http://schemas.microsoft.com/netfx/2009/xaml/activities/presentation" xmlns:scg="clr-namespace:System.Collections.Generic;assembly=System" xmlns:scg1="clr-namespace:System.Collections.Generic;assembly=System.ServiceModel" xmlns:scg2="clr-namespace:System.Collections.Generic;assembly=System.Core" xmlns:scg3="clr-namespace:System.Collections.Generic;assembly=mscorlib" xmlns:sd="clr-namespace:System.Data;assembly=System.Data" xmlns:sd1="clr-namespace:System.Data;assembly=System.Data.DataSetExtensions" xmlns:sl="clr-namespace:System.Linq;assembly=System.Core" xmlns:st="clr-namespace:System.Text;assembly=mscorlib" xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml">
  <local:TryCatchWorkflow.Implementation>
    <Sequence sad:XamlDebuggerXmlReader.FileName="d:\users\wxd\documents\visual studio 2010\Projects\ExceptionWorkflow\ExceptionWorkflow\TryCatchWorkflow.xaml">
      <TryCatch>
        <TryCatch.Try>
          <local:ExceptionActivity />
        </TryCatch.Try>
        <TryCatch.Catches>
          <Catch x:TypeArguments="s1:Exception">
            <ActivityAction x:TypeArguments="s1:Exception">
              <ActivityAction.Argument>
                <DelegateInArgument x:TypeArguments="s1:Exception" Name="exception" />
              </ActivityAction.Argument>
              <WriteLine Text="[exception.Message]" />
            </ActivityAction>
          </Catch>
        </TryCatch.Catches>
        <TryCatch.Finally>
          <WriteLine Text="A" />
        </TryCatch.Finally>
      </TryCatch>
      <TryCatch>
        <TryCatch.Try>
          <Throw Exception="[New ExceptionWorkflow.myException(&quot;wxd异常测试&quot;)]" />
        </TryCatch.Try>
        <TryCatch.Catches>
          <Catch x:TypeArguments="s1:Exception">
            <ActivityAction x:TypeArguments="s1:Exception">
              <ActivityAction.Argument>
                <DelegateInArgument x:TypeArguments="s1:Exception" Name="exception" />
              </ActivityAction.Argument>
              <WriteLine Text="[&quot;System.Exception 捕获:&quot; + exception.Message()]" />
            </ActivityAction>
          </Catch>
          <Catch x:TypeArguments="local:myException">
            <ActivityAction x:TypeArguments="local:myException">
              <ActivityAction.Argument>
                <DelegateInArgument x:TypeArguments="local:myException" Name="exception" />
              </ActivityAction.Argument>
              <WriteLine Text="[&quot;myException 捕获:&quot; + exception.Message()]" />
            </ActivityAction>
          </Catch>
        </TryCatch.Catches>
        <TryCatch.Finally>
          <WriteLine Text="B" />
        </TryCatch.Finally>
      </TryCatch>
    </Sequence>
  </local:TryCatchWorkflow.Implementation>
</local:TryCatchWorkflow>